File: homework/week2/2025320058-twuel0921/Stack.java
public class Stack<T> {
    private final int MAX_SIZE = 100;
    private int top = -1;
    private Object[] elemetArray;

    public Stack() {
        elemetArray = new Object[MAX_SIZE];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == MAX_SIZE-1;
    }

    public T peek() {
        return (T) elemetArray[top];
    }

    public T push(T item) {
        if(isFull()) {
            throw new ArrayIndexOutOfBoundsException("Full Stack!");
        }
        elemetArray[++top] = item;
        return item;
    }

    public T pop() {
        if(isEmpty()) {
            throw new ArrayIndexOutOfBoundsException("Empty Stack!");
        }
        return (T) elemetArray[top--];
    }
}

File: homework/week2/2025320609-joohyukpark/stack.c
#include <stdio.h>
#define MAX 100

int stack[MAX];
int top = -1;

void push(int x) {
    if (top < MAX - 1) {
        stack[++top] = x;
    }
}

int pop() {
    if (top >= 0) {
        return stack[top--];
    }
    return -1;
}

int peek() {
    if (top >= 0) {
        return stack[top];
    }
    return -1;
}

int is_empty() {
    return top == -1;
}

int is_full() {
    return top == MAX - 1;
}

int main() {
    push(10);
    push(20);
    printf("%d\n", pop());
    printf("%d\n", peek());
    return 0;
}
